src/java/org/apache/cassandra/index/SingletonIndexGroup.java

```diff
@@ -120,6 +120,25 @@ public Index.QueryPlan queryPlanFor(RowFilter rowFilter)
         return null;
     }
 
+    @Override
+    public Index.QueryPlan queryPlanForIndices(RowFilter rowFilter, Set<Index> indexes)
```

Comment:
This one looks like almost a duplicate of `Index.QueryPlan queryPlanFor`, which kinds of reinforces the idea that we could have a single signature for this.

Reply:
This one is wrong in my opinion. This class is about single index, we can never build plan of a subset which would be 0 indexes... I have it throwing unsupported in my next changes. But I agree we may want to combine into one method, though it may be a bit confusing and overloaded. 
As I mentioned - the current state is draft with main functionality for feedback - the actual polishing is in the making :-)